xs/src/libslic3r/PrintObject.cpp

```diff
@@ -695,13 +695,12 @@ PrintObject::_make_perimeters()
             const Flow ext_perimeter_flow       = layerm.flow(frExternalPerimeter);
             const coord_t ext_perimeter_width   = ext_perimeter_flow.scaled_width();
             const coord_t ext_perimeter_spacing = ext_perimeter_flow.scaled_spacing();
-            
             for (Surfaces::iterator slice = layerm.slices.surfaces.begin();
                 slice != layerm.slices.surfaces.end(); ++slice) {
                 while (true) {
                     // compute the total thickness of perimeters
                     const coord_t perimeters_thickness = ext_perimeter_width/2 + ext_perimeter_spacing/2
-                        + (region.config.perimeters-1 + region.config.extra_perimeters) * perimeter_spacing;
+                        + (region.config.perimeters-1 + region.config.extra_perimeters + slice->extra_perimeters) * perimeter_spacing;
```

Comment:
Oh, that's right!
And I clearly see my error at porting time: `region.config.extra_perimeters` is a boolean: it should not be summed in that expression!  I wrote `region.config.extra_perimeters` instead of `slice->extra_perimeters`.